### ISSUE TITLE
Introspect fails inside LazyVStack

### ISSUE THREAD

**Issue by hugehoge** [siteline/SwiftUI-Introspect#140]
## Environment

- Xcode 13.2.1
- iOS 15.2 simulator
- Introspect 0.1.4

### Code
```swift
struct ContentView: View {
  var body: some View {
    ScrollView {
      LazyVStack {
        ForEach(0..<20) { index in
          Color(hue: Double(index * 17 % 100) / 100, saturation: 0.5, brightness: 1.0)
            .frame(height: UIScreen.main.bounds.height / 10)
        }

        TextField("Placholder", text: .constant(""))
          .introspectTextField { view in
            view.layer.backgroundColor = UIColor.red.cgColor
          }
          .padding()

        Toggle("Toggle", isOn: .constant(true))
          .introspectSwitch { view in
            view.thumbTintColor = .red
          }
          .padding()
      }
    }
  }
}
```
### Result

|Result|If use VStack|
|-|-|
|<video src="https://user-images.githubusercontent.com/29995702/151662930-fa8ca2c3-0a48-4174-be63-1f2e71744c95.mp4">|<video src="https://user-images.githubusercontent.com/29995702/151662943-7488a9fa-3f25-4853-af4d-fef55887eafc.mp4">|

**Comment by hugehoge** [comment]
It seems that the following workaround would not work well in LazyVStack.

https://github.com/siteline/SwiftUI-Introspect/blob/72a509c93166540c0adf8323fd2652daade7f9f6/Introspect/UIKitIntrospectionView.swift#L47-L62

An alternative workaround I tried is below:

```diff
--- a/Introspect/UIKitIntrospectionView.swift
+++ b/Introspect/UIKitIntrospectionView.swift
@@ -5,6 +5,8 @@ import SwiftUI
 /// Introspection UIView that is inserted alongside the target view.
 @available(iOS 13.0, *)
 public class IntrospectionUIView: UIView {
+
+    var moveToWindowHandler: (() -> Void)?

     required init() {
         super.init(frame: .zero)
@@ -16,6 +18,11 @@ public class IntrospectionUIView: UIView {
     required init?(coder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
+
+    override public func didMoveToWindow() {
+        super.didMoveToWindow()
+        moveToWindowHandler?()
+    }
 }

 /// Introspection View that is injected into the UIKit hierarchy alongside the target view.
@@ -53,7 +60,7 @@ public struct UIKitIntrospectionView<TargetViewType: UIView>: UIViewRepresentabl
         _ uiView: IntrospectionUIView,
         context: UIViewRepresentableContext<UIKitIntrospectionView>
     ) {
-        DispatchQueue.main.async {
+        uiView.moveToWindowHandler = {
             guard let targetView = self.selector(uiView) else {
                 return
             }
```

Use [UIView.didMoveToWindow](https://developer.apple.com/documentation/uikit/uiview/1622527-didmovetowindow) instead of `DispatchQueue.main.async` to wait for the introspection view to be inserted to the hierarchy.

In my quick check, it would fix an issue that the use of introspect inside LazyVStack.

However, it makes some existing test cases failed and may cause many other effects.

<img width="488" alt="Test" src="https://user-images.githubusercontent.com/29995702/151663470-c97df3bb-082b-4008-b2e9-4aa7401f593c.png">

**Comment by plaps153** [comment]
Same here!

**Comment by hugehoge** [comment]
**UPDATE**

Leaving `DispatchQueue.main.async`, the existing tests will be passed.

```diff
 /// Introspection View that is injected into the UIKit hierarchy alongside the target view.
@@ -53,11 +70,13 @@ public struct UIKitIntrospectionView<TargetViewType: UIView>: UIViewRepresentabl
         _ uiView: IntrospectionUIView,
         context: UIViewRepresentableContext<UIKitIntrospectionView>
     ) {
-        DispatchQueue.main.async {
-            guard let targetView = self.selector(uiView) else {
-                return
+        uiView.moveToWindowHandler = {
+            DispatchQueue.main.async {
+                guard let targetView = self.selector(uiView) else {
+                    return
+                }
+                self.customize(targetView)
             }
-            self.customize(targetView)
         }
     }
 }
```

**Comment by davdroman** [comment]
Apparently fixed in #153. Will tag a release soon.
